### ISSUE TITLE
Revisit email length restriction

### ISSUE THREAD

**Issue by DFurnes** [DoSomething/northstar#355]
In response to #240, we added a length restriction on emails (specifically, that they have to be less than 60 characters long) because Drupal was refusing to register users with longer emails. Strangely, we're now getting validation errors when trying to migrate users from Drupal who have emails _longer_ than that!

I just tested registering a user with a whopping 76-character long email on staging and it seemed to work just fine, including signing up and reporting back for a campaign. This user, however, triggers a validation error when trying to forward it to Northstar.

```
loremipsumdolorsitametconsecteturadipiscingelitduisut1234567890b@example.com
```

Perhaps the 60-character restriction has been increased to 76 characters? 😕

**Comment by DFurnes** [comment]
It seems like both Drupal's [`valid_email_address`](https://api.drupal.org/api/drupal/includes%21common.inc/function/valid_email_address/7.x) and Laravel's built in [`email` validator](https://github.com/laravel/framework/blob/6a86d20f404d324956bc7b1a38f376265bde3cc7/src/Illuminate/Validation/Validator.php#L1466-L1476) both use PHP's `FILTER_VALIDATE_EMAIL` and so they _shooooould_ give similar results. That doesn't explain why Tong was seeing errors before, but perhaps the issue that was causing that problem has been fixed...?

@mshmsh5000 @angaither Do you think it's safe to remove this restriction?

**Comment by mshmsh5000** [comment]
Yyyyeaaaaahhhhh that's confusing. And looking back, I see we were talking about the length limit on the _username_, not the email address.

So the error could be if there were logic in Phoenix's user registration logic that created a default username based on the email address. But it only ever seems to use the UID:

https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_user/dosomething_user.module#L860

For instance, the username of your super-long email test above is `1707247`.

It looks safe on paper to remove this restriction.

**Comment by DFurnes** [comment]
Added a fix in #356. Luckily it doesn't look like many "real" accounts were affected... because honestly like who has time for a 60+ character email address? 😆 For the most part it's all old Facebook Connect "proxy" email addresses.
